### PR TITLE
EN 1131 user role detail

### PR DESCRIPTION
**Pull request by mcarta-rt**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: src/state/roles/actions.js [hunk 1]
```diff
@@ -55,6 +75,21 @@ export const fetchRole = roleCode => dispatch =>
     });
   });
 
+export const fetchRoleDetail = roleCode => dispatch =>
+  new Promise((resolve) => {
+    getRole(roleCode).then((response) => {
+      response.json().then((data) => {
+        if (response.ok) {
+          dispatch(setSelected(data.payload));
+          resolve();
```
Comment (alessioscalici) [inline src/state/roles/actions.js]: You can call `resolve()` only once, after the if/else block

File: src/state/roles/actions.js [hunk 2]
```diff
@@ -100,3 +135,21 @@ export const sendDeleteRole = roleCode => dispatch =>
...
+
+export const fetchUserRefs = (roleCode, page = { page: 1, pageSize: 10 }, params = '') => dispatch =>
+  new Promise((resolve) => {
+    dispatch(toggleLoading('references'));
+    getUserReferences(roleCode, page, params).then((response) => {
+      response.json().then((data) => {
+        if (response.ok) {
+          dispatch(setUserRefs(data.payload));
+          dispatch(setPage(data.metaData));
+          resolve();
```
Comment (alessioscalici) [inline src/state/roles/actions.js]: You can call `resolve()` only once, after the if/else block, after the last dispatch

File: src/state/roles/selectors.js [hunk 3]
```diff
@@ -1,10 +1,24 @@
 import { createSelector } from 'reselect';
+import { getPermissionsList } from 'state/permissions/selectors';
 
 export const getRoles = state => state.roles;
 export const getRolesIdList = state => state.roles.list;
 export const getRolesMap = state => state.roles.map;
+export const getSelectedRolePermissions = state => state.roles.selected.permissions;
```
Comment (alessioscalici) [inline src/state/roles/selectors.js]: `selected` can be null/undefined, these selectors are unsafe. Please use `state.roles.selected && state.roles.selected.permissions`, or `get(state.roles, 'selected.permissions')` (`get` is a function from lodash)

File: test/state/permissions/actions.test.js [hunk 4]
```diff
@@ -41,17 +41,18 @@ describe('state/permissions/actions', () => {
...
         expect(actions[2]).toHaveProperty('type', SET_PAGE);
+        expect(actions[3]).toHaveProperty('type', TOGGLE_LOADING);
         done();
       }).catch(done.fail);
     });
 
     it('permissions is defined and properly valued', (done) => {
       store.dispatch(fetchPermissions()).then(() => {
-        const actionPayload = store.getActions()[0].payload;
+        const actionPayload = store.getActions()[1].payload;
```
Comment (alessioscalici) [inline test/state/permissions/actions.test.js]: Please expect arrays "toHaveLength" before accessing them

File: test/ui/roles/detail/DetailRolePage.test.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  it('renders without crashing', () => {
+    expect(component.exists()).toEqual(true);
+  });
+
+  it('verify if exist InternalPage with class DetailRolePage', () => {
+    expect(component.find('InternalPage').hasClass('DetailRolePage')).toEqual(true);
+  });
+
+  it('verify if has a breadcrumb', () => {
+    expect(component.find('Breadcrumb').exists());
```
Comment (alessioscalici) [inline test/ui/roles/detail/DetailRolePage.test.js]: This isn't actually testing anything. The correct test would be

`expect(component.find('Breadcrumb')).toExist();`

or

`expect(component.find('Breadcrumb').exists()).toBe(true);`

File: test/ui/roles/detail/DetailRolePage.test.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  });
+
+  it('verify if has a page title', () => {
+    expect(component.find('PageTitle').exists());
```
Comment (alessioscalici) [inline test/ui/roles/detail/DetailRolePage.test.js]: Here too

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by alessioscalici** [comment]
Please retest this

**Comment by alessioscalici** [comment]
retest this please

**Comment by mcarta-rt** [comment]
retest this please

**Comment by mcarta-rt** [comment]
retest this please

**Comment by mcarta-rt** [comment]
retest this please

**Comment by alessioscalici** [comment]
retest this please

**Comment by pietrangelo** [comment]
retest this please

**Comment by alessioscalici** [comment]
retest this please

**Comment by pietrangelo** [comment]
retest this please

**Comment by mcarta-rt** [comment]
retest this please

**Pull request closed without merging**